src/Tests/Microsoft.DotNet.ApiCompatibility.Tests/Rules/EnumsMustMatchTests.cs

```diff
@@ -0,0 +1,160 @@
+using Xunit;
+
+namespace Microsoft.DotNet.ApiCompatibility.Tests
+{
+    public class EnumsMustMatchTests
+    {
+        [Fact]
+        public static void DifferencesReported()
+        {
+            string leftSyntax = @"
+namespace CompatTests
+{
+  public enum First {
+    A = 0,
+    B = 1,
+    C = 2,
+    D = 3,
+  }
+}
+";
+
+            string rightSyntax = @"
+namespace CompatTests
+{
+  public enum First {
+    E = 4,
+    D = 3,
+    C = 2,
+    B = 1,
+    A = 1,
+  }
+}
+";
+            IAssemblySymbol left = SymbolFactory.GetAssemblyFromSyntax(leftSyntax);
+            IAssemblySymbol right = SymbolFactory.GetAssemblyFromSyntax(rightSyntax);
+            ApiComparer differ = new();
+            IEnumerable<CompatDifference> differences = differ.GetDifferences(new[] { left }, new[] { right });
+            CompatDifference[] expected = new[]
+            {
+                new CompatDifference(DiagnosticIds.EnumValuesMustMatch, string.Empty, DifferenceType.Changed, "F:CompatTests.First.A"),
+            };
+            Assert.Equal(expected, differences);
+        }
+
+        [Fact]
+        public static void RemovedEnum()
+        {
+            string leftSyntax = @"
+namespace CompatTests
+{
+  public enum First {
+    A = 0,
+    B = 1,
+    C = 2,
+    D = 3,
+  }
+  public enum Second {}
+}
+";
+
+            string rightSyntax = @"
+namespace CompatTests
+{
+  public enum First {
+    D = 3,
+    C = 2,
+    B = 1,
+    A = 0,
+  }
+}
+";
+            IAssemblySymbol left = SymbolFactory.GetAssemblyFromSyntax(leftSyntax);
+            IAssemblySymbol right = SymbolFactory.GetAssemblyFromSyntax(rightSyntax);
+            ApiComparer differ = new();
+            IEnumerable<CompatDifference> differences = differ.GetDifferences(new[] { left }, new[] { right });
+            Assert.NotEmpty(differences);
+        }
+
+        [Fact]
+        public static void AddedEnum()
+        {
+            string leftSyntax = @"
+namespace CompatTests
+{
+  public enum First {
+    A = 0,
+    B = 1,
+    C = 2,
+    D = 3,
+  }
+}
+";
+
+            string rightSyntax = @"
+namespace CompatTests
+{
+  public enum First {
+    D = 3,
+    C = 2,
+    B = 1,
+    A = 0,
+  }
+  public enum Second {}
+}
+";
+            IAssemblySymbol left = SymbolFactory.GetAssemblyFromSyntax(leftSyntax);
+            IAssemblySymbol right = SymbolFactory.GetAssemblyFromSyntax(rightSyntax);
+            ApiComparer differ = new();
+            IEnumerable<CompatDifference> differences = differ.GetDifferences(new[] { left }, new[] { right });
+            Assert.Empty(differences);
+        }
+
+        [Fact]
+        public static void BackingStoreChanged()
+        {
+            string leftSyntax = @"
+namespace CompatTests
+{
+  public enum First: short {
+    A = 0,
+    B = 1,
+    C = 2,
+    D = 3,
+  }
+}
+";
+
+            string rightSyntax = @"
+namespace CompatTests
+{
+  public enum First: int {
```

Comment:
nit: the inheritance operator is usually separated by a whitespace.

```suggestion
  public enum First : int {
```